specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualNetworks.json

```diff
@@ -0,0 +1,575 @@
+      "get": {
+        "operationId": "virtualnetworks_ListBySubscription",
+        "x-ms-examples": {
+          "ListVirtualNetworkBySubscription": {
+            "$ref": "./examples/ListVirtualNetworkBySubscription.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Lists all virtualnetworks under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/virtualnetworksListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "virtualnetworksProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "networkType": {
+          "type": "string",
+          "description": "Type of the network"
+        },
+        "subnets": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "Name - The name of the resource that is unique within a resource group. This name can be used to access the resource."
+              },
+              "properties": {
+                "type": "object",
+                "x-ms-client-flatten": true,
+                "properties": {
+                  "addressPrefix": {
+                    "type": "string",
+                    "description": "Cidr for this subnet - IPv4, IPv6"
+                  },
+                  "addressPrefixes": {
+                    "type": "array",
+                    "items": {
+                      "type": "string"
+                    },
+                    "description": "AddressPrefixes - List of address prefixes for the subnet."
+                  },
+                  "ipAllocationMethod": {
+                    "type": "string",
+                    "description": "IPAllocationMethod - The IP address allocation method. Possible values include: 'Static', 'Dynamic'"
+                  },
+                  "ipConfigurationReferences": {
+                    "type": "array",
+                    "items": {
+                      "type": "object",
+                      "properties": {
+                        "ID": {
```

Comment:
>ID [](http://example.com/codeflow?start=25&length=2)

Suggest resourceId. At minimum it should be camelCase (== id)

Reply:
We expect name of the resource here. Updated the description.